hal/armv7m/cpu.c

```diff
@@ -255,6 +255,23 @@ void hal_wdgReload(void)
 }
 
 
+/* cache management */
+
+
+void hal_cleanDCache(ptr_t start, size_t len)
+{
+	(void)start;
+	(void)len;
+
+#ifdef CPU_IMXRT
+	/* Clean the whole cache, no harm in being overzealous */
+	_imxrt_cleanDCache();
+#else
+	/* TODO */
+#endif
```

Comment:
Data cache cleaning for Cortex-M is the same for `imxrt` and `stm32`. I think that the `ifdef` is not needed here and the code can be common for all Cortex-M7 architectures.

Reply:
Can't really be done due to code arch - scb address is kept in x_common structs per arch

Reply:
If you look at functions using `scb` registers in `imxrt106x`, `imxrt117x` and `stm32`, they are the same. There is copy-paste code between 3 targets.

Reply:
Yeah, I wanted to raise a topic to avoid copy-pasting in the future. Maybe put a `FIXME` comment to not forget about it.